src/main/java/uk/gov/hmcts/reform/finrem/caseorchestration/handler/AmendApplicationContestedAboutToSubmitHandler.java

```diff
@@ -0,0 +1,326 @@
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_DOMESTIC_VIOLENCE_CHECKLIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_EXEMPTIONS_CHECKLIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_OTHER_GROUNDS_CHECKLIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_OTHER_GROUNDS_TEXTBOX;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_PREVIOUS_ATTENDANCE_CHECKLIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_PREVIOUS_ATTENDANCE_TEXTBOX;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_URGENCY_CHECKLIST;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MIAM_URGENCY_TEXTBOX;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.MINI_FORM_A;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_ADDRESS;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_EMAIL;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.RESPONDENT_PHONE;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.TYPE_OF_APPLICATION;
+import static uk.gov.hmcts.reform.finrem.caseorchestration.model.ccd.CCDConfigConstant.TYPE_OF_APPLICATION_DEFAULT_TO;
+
+
+@Slf4j
+@Service
+public class AmendApplicationContestedAboutToSubmitHandler implements CallbackHandler {
+
+    private final OnlineFormDocumentService onlineFormDocumentService;
+    private final CaseFlagsService caseFlagsService;
+
+    private static final String DIVORCE_STAGE_REACHED = "divorceStageReached";
+    private static final String DIVORCE_UPLOAD_EVIDENCE_2 = "divorceUploadEvidence2";
+    private static final String DIVORCE_DECREE_ABSOLUTE_DATE = "divorceDecreeAbsoluteDate";
+    private static final String DIVORCE_UPLOAD_PETITION = "divorceUploadPetition";
+    private static final String DIVORCE_UPLOAD_EVIDENCE_1 = "divorceUploadEvidence1";
+    private static final String DIVORCE_DECREE_NISI_DATE = "divorceDecreeNisiDate";
+
+    public AmendApplicationContestedAboutToSubmitHandler(OnlineFormDocumentService onlineFormDocumentService, CaseFlagsService caseFlagsService) {
+        this.onlineFormDocumentService = onlineFormDocumentService;
+        this.caseFlagsService = caseFlagsService;
+    }
+
+    @Override
+    public boolean canHandle(CallbackType callbackType, CaseType caseType, EventType eventType) {
+        return CallbackType.ABOUT_TO_SUBMIT.equals(callbackType)
+            && CaseType.CONTESTED.equals(caseType)
+            && (EventType.AMEND_CONTESTED_APP_DETAILS.equals(eventType));
```

Comment:
I think this is missing an event type. It needs to handle FR_amendApplication and FR_amendPaperApplication

Reply:
Thanks, committed here [0c16b58](https://github.com/hmcts/finrem-case-orchestration-service/pull/1772/commits/0c16b583eb86b6b0dc1717ad4ec4de46c87716c0)